src/NewTools-Debugger-Tests/StDebuggerConfigurationCommandTreeBuilderTest.class.st

```diff
@@ -0,0 +1,47 @@
+Class {
+	#name : #StDebuggerConfigurationCommandTreeBuilderTest,
+	#superclass : #StDebuggerToolCommandTreeBuilderTest,
+	#category : #'NewTools-Debugger-Tests-Model'
+}
+
+{ #category : #'instance creation' }
+StDebuggerConfigurationCommandTreeBuilderTest >> newBuilder [
+
+	^ StDebuggerConfigurationToolbarCommandTreeBuilder
+		  buildCommandsGroupWith: debugger
+		  forRoot: rootGroup
+]
+
+{ #category : #tests }
+StDebuggerConfigurationCommandTreeBuilderTest >> testConfigurationCommands [
+	self
+		assertCommandGroup: builder generalConfigurationCommandsGroup
+		hasSameCommands: builder configurationCommandClasses
+		withContext: debugger
+		displayStrategy: CmUIDisplayAsGroup
+]
+
+{ #category : #tests }
+StDebuggerConfigurationCommandTreeBuilderTest >> testConfigurationCommandsClasses [
+
+	self assertCollection: builder configurationCommandClasses equals: {
+			StDebuggerToggleFilterStackCommand.
+			StDebuggerSettingsCommand }
```

Comment:
So, if we add a class, the test should break ? Looks weird

Reply:
Yes, the test is a specification, it is meant to tell what the default debugger commands are.
It means that if we change the default commands, we get notified of it because the test breaks.
We then have to decide if that is a bug (then fix it) or if the command should change on purpose and therefore update the test.

It is not elegant, but it facilitates maintenance compared to using pragmas as in the previous implementation.
The alternative using pragmas would be to have lots of pragmas, each of them representing a particular group of commands.